cpp/src/arrow/io/interfaces.h

```diff
@@ -129,10 +129,10 @@ class ARROW_EXPORT WriteableFileInterface : public OutputStream, public Seekable
   WriteableFileInterface() { set_mode(FileMode::READ); }
 };
 
-class ARROW_EXPORT ReadWriteFileInterface : public ReadableFileInterface,
+class ARROW_EXPORT ReadWriteFileInterface : public RandomAccessFile,
                                             public WriteableFileInterface {
  protected:
-  ReadWriteFileInterface() { ReadableFileInterface::set_mode(FileMode::READWRITE); }
+  ReadWriteFileInterface() { RandomAccessFile::set_mode(FileMode::READWRITE); }
 };
```

Comment:
Would you mind adding a `using ReadableFileInterface = RandomAccessFile;` to ease matters for thirdparty users (this class is used in parquet-cpp, for example)